### PR TITLE
Define a Realm Object Schema

### PR DESCRIPTION
**Pull request by nathan-contino-mongo**
### Docs staging link (requires sign-in on MongoDB Corp SSO):
https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/define-a-realm-object-schema/sdk/android/examples/define-a-realm-object-model

### CODE REVIEW

File: source/sdk/android/examples/define-a-realm-object-model.txt [hunk 1]
```diff
@@ -12,57 +13,642 @@ Define a Realm Object Schema - Android SDK
...
+- ``RealmObject``
+- ``RealmList <? extends RealmObject>``
+
+Use annotations to indicate whether a given field represents a foreign
+key relationship or an embedded object relationship. For more
+information, see :ref:`Relationship Annotations <android-relationships>`.
+
+.. seealso::
+
+   :ref:`Relationships <android-client-relationships>`
```
Comment (terakilobyte) [inline source/sdk/android/examples/define-a-realm-object-model.txt:91]: issue: This appears duplicated

File: source/sdk/android/examples/define-a-realm-object-model.txt [hunk 2]
```diff
@@ -12,57 +13,642 @@ Define a Realm Object Schema - Android SDK
...
+.. _android-field-relationships:
 
 Define a Relationship Property
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
+{+client-database+} supports several kinds of object relationships.
+
+.. seealso::
+
+   :ref:`Relationships <android-client-relationships>`
```
Comment (terakilobyte) [inline source/sdk/android/examples/define-a-realm-object-model.txt:103]: issue: This appears duplicated

File: source/sdk/android/examples/define-a-realm-object-model.txt [hunk 3]
```diff
@@ -12,57 +13,642 @@ Define a Realm Object Schema - Android SDK
...
+
+.. _android-field-relationships-many-to-one:
+
+Many-to-One
+===========
+
+To set up a many-to-one or one-to-one relationship, create a field
+with whose type is a {+service-short+} object in your application:
```
Comment (terakilobyte) [inline source/sdk/android/examples/define-a-realm-object-model.txt:111]:
```suggestion
whose type is a {+service-short+} object in your application:
```

File: source/sdk/android/examples/define-a-realm-object-model.txt [hunk 4]
```diff
@@ -12,57 +13,642 @@ Define a Realm Object Schema - Android SDK
...
+.. _android-field-relationships-embedded:
 
 Define an Embedded Object Property
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
+{+client-database+} provides the ability to nest objects within other
+objects. This has several advantages:
+
+- If using {+sync+}, objects will translate into MongoDB documents that
+  follow a :manual:`denormalized data model </manual/core/data-modeling-introduction/>`.
```
Comment (terakilobyte) [inline source/sdk/android/examples/define-a-realm-object-model.txt:226]: issue: this link doesn't work

Comment (terakilobyte) [inline source/sdk/android/examples/define-a-realm-object-model.txt:226]:
```suggestion
  follow a :manual:`denormalized data model </core/data-modeling-introduction/>`.
```

File: source/sdk/android/examples/define-a-realm-object-model.txt [hunk 5]
```diff
@@ -12,57 +13,642 @@ Define a Realm Object Schema - Android SDK
...
+
+      In Kotlin, fields are considered nullable only if a field is
+      marked nullable with the Kotlin `? operator
+      <https://kotlinlang.org/docs/reference/null-safety.html>`__ except
+      for the following types:
+
+      - ``String``
+      - ``Date``
+      - ``ObjectId``
+      - ``Decimal128``
```
Comment (terakilobyte) [inline source/sdk/android/examples/define-a-realm-object-model.txt:383]: question: this appears to either contradict or supplement the previous warning about Kotlin types and nullability. Can they be clarified and combined?

File: source/sdk/android/examples/define-a-realm-object-model.txt [hunk 6]
```diff
@@ -12,57 +13,642 @@ Define a Realm Object Schema - Android SDK
...
+      - ``ObjectId``
+      - ``Integer``
+      - ``Long``
+      - ``Short``
+      - ``Byte`` or ``byte[]``
+      - ``Boolean``
+      - ``Float``
+      - ``Double``
+
+Primitive types like ``int`` and the ``RealmList`` type are non-nullable
```
Comment (terakilobyte) [inline source/sdk/android/examples/define-a-realm-object-model.txt:406]: nitpick: hard to quickly parse.

When I first read this I read it as `RealmList` is a primitive, which surprised me. I looked into it more and there don't appear to be any "primitive" types in Kotlin, do you mean ["basic" types](https://kotlinlang.org/docs/reference/basic-types.html)

 Maybe something like:

All basic types (``Numbers``, ``Strings``, etc...), and the ``RealmList`` type, are non-nullable...

File: source/sdk/android/examples/define-a-realm-object-model.txt [hunk 7]
```diff
@@ -12,57 +13,642 @@ Define a Realm Object Schema - Android SDK
...
+
+.. _android-default-field-values:
+
+Default Field Values
+~~~~~~~~~~~~~~~~~~~~
+
+To assign a default value to a field, use the built-in language features
+to assign default values. In Kotlin, you can assign a default value in
+the field declaration. In Java, use the class constructor to assign
+default values.
```
Comment (terakilobyte) [inline source/sdk/android/examples/define-a-realm-object-model.txt:418]: nitpick: required reading.

It seems this paragraph could be decomposed to its core components and then distributed into the language tabs. That way someone who has been following along with Kotlin doesn't even see the mention of how to do it in Java

**Review by terakilobyte — CHANGES_REQUESTED** [review]
Solid content, a few issues to address and some nits to think about.

**Review by nathan-contino-mongo — COMMENTED** [review]
@terakilobyte thanks for the review, implemented updates to address your feedback. Please take a look when you get the chance!

**Review by terakilobyte — APPROVED** [review]
lgtm

**Pull request merged**